code/datums/components/food/customizable.dm

```diff
@@ -0,0 +1,225 @@
+/**
+ * # Custom Food Component
+ *
+ * When added to an atom, edible ingredients can be put into that.
+ * The sprite is updated and reagents are transfered.
+ *
+ * If the component is added to something that is processed, creating new objects (being cut, for example),
+ * the replacement type needs to also have the component. The ingredients will be copied over. Reagents are not
+ * copied over since edible components already take care of that.
+ */
+/datum/component/customizable
```

Comment:
This is a pretty generic name for something so specific--should probably be renamed to customizable_food, or something.